src/LiveComponent/src/Util/QueryStringPropsExtractor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Util;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\UX\LiveComponent\Exception\HydrationException;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LivePropMetadata;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+final class QueryStringPropsExtractor
+{
+    public function __construct(private readonly LiveComponentHydrator $hydrator)
+    {
+    }
+
+    /**
+     * Extracts relevant query parameters from the current URL and hydrates them.
+     */
+    public function extract(Request $request, LiveComponentMetadata $metadata, object $component): array
+    {
+        $query = $request->query->all();
+
+        if (empty($query)) {
+            return [];
+        }
+        $data = [];
+
+        foreach ($metadata->getAllLivePropsMetadata() as $livePropMetadata) {
+            if ($queryStringMapping = $livePropMetadata->getQueryStringMapping()) {
+                if (null !== ($value = $query[$queryStringMapping['name']] ?? null)) {
+                    if (\is_array($value) && $this->isNumericIndexedArray($value)) {
+                        // Sort numeric array
+                        ksort($value);
```

Comment:
I made a few tests, and i don't think we should.

Or there will be unnecessary changes (if per example i had [2 => 2, 1 => 1] in the liveprop value)